code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm

```diff
@@ -24,77 +83,153 @@
-		to_chat(X, "<span class='xenodanger'>Pouncing from the shadows, we stagger our victim.</span>")
+		to_chat(owner, "<span class='xenodanger'>Pouncing from the shadows, we stagger our victim.</span>")
 
-/datum/action/xeno_action/activable/pounce/hunter/sneak_attack()
-	var/mob/living/carbon/xenomorph/hunter/X = owner
-	if(X.can_sneak_attack) //If we could sneak attack, add a cooldown to sneak attack
-		to_chat(X, "<span class='xenodanger'>Our pounce has left us off-balance; we'll need to wait [HUNTER_POUNCE_SNEAKATTACK_DELAY*0.1] seconds before we can Sneak Attack again.</span>")
-		X.can_sneak_attack = FALSE
-		addtimer(CALLBACK(X, /mob/living/carbon/xenomorph/hunter/.proc/sneak_attack_cooldown), HUNTER_POUNCE_SNEAKATTACK_DELAY)
+/datum/action/xeno_action/stealth/proc/sneak_attack_slash(datum/source, mob/living/target, damage, list/damage_mod, list/armor_mod)
+	if(!stealth || !can_sneak_attack)
+		return
 
-/mob/living/carbon/xenomorph/hunter/proc/sneak_attack_cooldown()
-	if(!stealth || can_sneak_attack)
+	var/staggerslow_stacks = 2
+	var/flavour
+	if(owner.m_intent == MOVE_INTENT_RUN && ( owner.last_move_intent > (world.time - HUNTER_SNEAK_ATTACK_RUN_DELAY) ) ) //Allows us to slash while running... but only if we've been stationary for awhile
+		flavour = "vicious"
+	else
+		armor_mod += HUNTER_SNEAK_SLASH_ARMOR_PEN
+		staggerslow_stacks *= 2
+		flavour = "deadly"
+
+	owner.visible_message("<span class='danger'>\The [owner] strikes [target] with [flavour] precision!</span>", \
+	"<span class='danger'>We strike [target] with [flavour] precision!</span>")
+	target.adjust_stagger(staggerslow_stacks)
+	target.add_slowdown(staggerslow_stacks)
+	
+	cancel_stealth()
+
+/datum/action/xeno_action/stealth/proc/sneak_attack_disarm(datum/source, mob/living/target, tackle_pain, list/pain_mod)
+	if(!stealth || !can_sneak_attack)
 		return
-	can_sneak_attack = TRUE
-	to_chat(src, "<span class='xenodanger'>We're ready to use Sneak Attack while stealthed.</span>")
-	playsound(src, "sound/effects/xeno_newlarva.ogg", 25, 0, 1)
+
+	var/staggerslow_stacks = 2
+	var/flavour
+	if(owner.m_intent == MOVE_INTENT_RUN && ( owner.last_move_intent > (world.time - HUNTER_SNEAK_ATTACK_RUN_DELAY) ) ) //Allows us to slash while running... but only if we've been stationary for awhile
+		pain_mod += 1.75
+		flavour = "vicious"
+	else
+		pain_mod += 3.5
+		staggerslow_stacks *= 2
+		flavour = "deadly"
+
+	owner.visible_message("<span class='danger'>\The [owner] strikes [target] with [flavour] precision!</span>", \
+	"<span class='danger'>We strike [target] with [flavour] precision!</span>")
+	target.ParalyzeNoChain(1.5 SECONDS)
+	target.adjust_stagger(staggerslow_stacks)
+	target.add_slowdown(staggerslow_stacks)
+	
+	cancel_stealth()
+
+/datum/action/xeno_action/stealth/proc/damage_taken(mob/living/carbon/xenomorph/X, damage_taken)
+	if(damage_taken > 15)
+		cancel_stealth()
+
+/datum/action/xeno_action/stealth/proc/plasma_regen(datum/source, list/plasma_mod)
+	handle_stealth()
```

Comment:
This doesn't actually get hooked up.

Without this the hunter can never get "standing still" stealth since the only other thing to trigger `handle_stealth()` is atom_moved.